File: src/QFSvb/QFSvb/FileSystem/FSElement.vb
Namespace QFSvb.FileSystem

    Public MustInherit Class FSElement

        Private iId As Integer
        Private sName As String

        'Constructor por defecto
        Sub New()

        End Sub

        Sub New(ByVal idFSElement As Integer)
            Me.iId = idFSElement
        End Sub

        ''' <summary>
        ''' 
        ''' </summary>
        ''' <value></value>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Public ReadOnly Property idFSElement()
            Get
                idFSElement = Me.iId
            End Get
        End Property

        Public Property name()
            Get
                name = Me.sName
            End Get
            Set(ByVal sValue)
                Me.sName = sValue
            End Set
        End Property

    End Class

End Namespace

File: src/QFS/QFS/FileSystem/FSElement.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace QFS.FileSystem
{
    abstract class FSElement
    {
        public FSElement() { }

        public FSElement(int id, String name) {
            this.iId = id;
            this.sName = name;
        }

        public FSElement(int id)
        {
            this.iId = id;
         
        }

        private int iId;
        private String sName;

        public readonly int id
        {
            get { return ciudad; }
        }

        public String name
        {
            get { return this.sName; }
            set { this.sName = value; } 
        }


    }
}
